frontend/src/state/start-controls.js

```diff
@@ -0,0 +1,41 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+export const timers = ["Fast", "Moderate", "Slow", "Leisurely"];
+
+export const initialState = {
+  addBots: true,
+  shufflePlayers: true,
+  useTimer: true,
+  timerLength: "Moderate", // Fast Moderate or Slow
+};
+
+const startControls = createSlice({
+  name: "startControls",
+  initialState,
+  reducers: {
+    toggleBots: state => {
+      state.addBots = !state.addBots;
+    },
+    toggleShufflePlayers: state => {
+      state.shufflePlayers = !state.shufflePlayers;
+    },
+    toggleUseTimer: state => {
+      state.useTimer = !state.useTimer;
+    },
+    setTimerLength: (state, action) => {
+      if (timers.includes(action.payload)) {
+        state.timerLength = action.payload;
+      }
+    }
+  }
+}
+);
```

Comment:
These reducers are what we could easily write tests over. Haven't seen this createSlice  thing before but we just need a way to reach in and check those reduce functions. These ones are simple, but would be good to get into habit of putting tests over ones which have multiple paths over them.

I would say we should have a testing pattern in place before merging this. Would set us up well. Happy to help